test/bucket.ts

```diff
@@ -62,24 +64,32 @@ let requestOverride;
 function fakeRequest() {
   return (requestOverride || requestCached).apply(null, arguments);
 }
-(fakeRequest as any).defaults = () => {
-  // Ignore the default values, so we don't have to test for them in every API
-  // call.
-  return fakeRequest;
-};
-(fakeRequest as any).get = (...args) => {
-  return (requestOverride.get || fakeRequest).apply(null, args);
-};
-(fakeRequest as any).head = (...args) => {
-  return (requestOverride.head || fakeRequest).apply(null, args);
-};
+extend(fakeRequest, {
```

Comment:
same feedback as above.  Using the lack of type safety on extend feels like a hack, and I'd rather define an interface.

Reply:
https://github.com/googleapis/nodejs-storage/pull/307#discussion_r203872303